app/org/sagebionetworks/bridge/dao/HealthDataDao.java

```diff
@@ -29,6 +29,15 @@
     int deleteRecordsForHealthCode(@Nonnull String healthCode);
 
     /**
+     * DAO method used by worker apps to fetch a health data record by the record ID.
+     *
+     * @param id
+     *         record ID
+     * @return health data record
+     */
+    HealthDataRecord getRecordById(@Nonnull String id);
```

Comment:
There's getRecordsForUploadDate(), so I wanted to contrast with getRecordById(). Plus, the HTTP APIs are /healthdata/byId/:id and /healthdata/byUploadDate/:uploadDate, so the DAO API matches the HTTP API.